packages/fxa-settings/src/lib/channels/firefox.ts

```diff
@@ -298,6 +311,42 @@ export class Firefox extends EventTarget {
   fxaCanLinkAccount(options: FxACanLinkAccount) {
     this.send(FirefoxCommand.Login, options);
   }
+
+  async requestSignedInUser(context: string) {
+    return Promise.race([
```

Comment:
I found a `Promise.race` a little more clear to understand what was happening instead of the OutstandingRequests class in #15723. If we have to do this a bunch in the future or there's some complexity we need to account for I think it could make sense to use in a follow up though.